der/derive/src/sequence.rs

```diff
@@ -127,35 +130,41 @@ impl SequenceField {
         });
 
         let attrs = FieldAttrs::parse(&field.attrs, type_attrs);
+        let field_type = field.ty.clone();
 
         if attrs.tag_mode == TagMode::Implicit {
             abort!(ident, "IMPLICIT tagging not supported for `Sequence`");
         }
 
-        Self { ident, attrs }
+        Self {
+            ident,
+            attrs,
+            field_type,
+        }
     }
 
     /// Derive code for decoding a field of a sequence.
     fn to_decode_tokens(&self) -> TokenStream {
         let ident = &self.ident;
-        if self.attrs.asn1_type.is_some() {
-            let field_decoder = self.attrs.decoder();
-            quote! { let #ident = #field_decoder.try_into()?; }
+        let ty = self.field_type.clone();
+        if None != self.attrs.default {
+            let fname = syn::Ident::new(&self.attrs.default.clone().unwrap(), self.ident.span());
+            quote!(let mut #ident = Some(decoder.decode::<#ty>()?.unwrap_or_else(|| #fname()));)
```

Comment:
Slightly more idiomatic way to write this would be something like:

```suggestion
        if let Some(default) = &self.attrs.default {
            let fname = syn::Ident::new(default, self.ident.span());
            quote!(let mut #ident = Some(decoder.decode::<#ty>()?.unwrap_or_else(|| #fname()));)
```

Reply:
Refactored per the suggestion in this spot and one other similar spot.